Source/Diagnostics/ParticleIO.cpp

```diff
@@ -299,3 +301,217 @@ storePhiOnParticles ( PinnedMemoryParticleContainer& tmp,
         }
     }
 }
+
+namespace
+{
+// This template is isolated in a separate namespace to simplify the usage of
+// the doDirectGatherVectorField template and avoid branching
+template <int depos_order, bool galerkin_interpolation>
+void
+storeEMFieldsOnParticles_t (PinnedMemoryParticleContainer& tmp,
+                            ElectromagneticSolverAlgo electromagnetic_solver_id,
+                            const bool fields_to_plot[],
+                            bool is_full_diagnostic)
```

Comment:
```suggestion
                            const bool is_full_diagnostic)
```
`const` can be used here.